sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/Plan.java

```diff
@@ -21,6 +22,8 @@
  * list of one or more error messages to be shown to the user,
  */
 public interface Plan extends ParentElement<Phase> {
+    void setErrors(List<String> errors);
```

Comment:
Feels weird adding a setter to this interface. How about having DefaultScheduler just create a new plan with errors pre-populated in that plan? The plan could omit the other pieces.

Something like:
```
final List<String> errors = new ArrayList<>();
errors.add("err: bad config");
return new Plan() {
  public List<String> getErrors() { return errors; }
};
```